### PR TITLE
finished task 11

### PR DESCRIPTION
**Pull request by NicholasF13**
Finsished task 11 - /api/articles (topic query) endpoint

### CODE REVIEW

File: app.test.js [hunk 1]
```diff
@@ -395,4 +395,37 @@ test("Returns users with the correct format", () => {
...
+  })
+  test("returns 400 and error message for invalid topic query parameter", () => {
+
+    return request(app)
+      .get(`/api/articles?topic=football`)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("Topic not found");
+      })
+  })
```
Comment (NRMSMITH) [inline app.test.js:440]: Wht happens when a topic exists (paper) but it has no associated articles? (hint, it should be a 200 and just send back an empty array!)

File: models/articles-model.js [hunk 2]
```diff
@@ -35,10 +43,25 @@ function selectArticles(){
     articles.votes,
     articles.article_img_url
     ORDER BY
-    articles.created_at DESC;`)
+    articles.created_at DESC`
+
+  let queryValues;
+
+    if (topic) {
```
Comment (NRMSMITH) [inline models/articles-model.js:57]: I think we should check beforehand if the topic exists - perhaps by using a greenlist of all the topics in the database that can be updated dynamically? 😄

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
A test to add and a greenlist to make 👍

File: app.test.js [hunk 3]
```diff
@@ -395,4 +395,47 @@ test("Returns users with the correct format", () => {
...
+          expect(article).toHaveProperty("article_id", expect.any(Number))
+          expect(article).toHaveProperty("topic", 'mitch')
+          expect(article).toHaveProperty("created_at", expect.any(String))
+          expect(article).toHaveProperty("votes", expect.any(Number))
+          expect(article).toHaveProperty("article_img_url", expect.any(String))
+          expect(article).toHaveProperty("comment_count", expect.any(String))
+        })
+      })
+  })
+  test("returns a status of 200 and an empty array if passed a topic with no associated articles", () => {
```
Comment (pippaamy) [inline app.test.js:422]: great test

**Review by pippaamy — COMMENTED** [review]
great changes, go ahead and merge!

**Pull request merged**